src/openforms/forms/tests/variables/test_viewset.py

```diff
@@ -205,6 +207,83 @@ def test_list_form_variables(self):
         self.assertIn(form_variable1.key, variables_keys)
         self.assertIn(form_variable2.key, variables_keys)
 
+    def test_list_form_variables_source_filter(self):
+        user = SuperUserFactory.create()
+        self.client.force_authenticate(user)
+
+        form = FormFactory.create()
+        FormStepFactory.create(
+            form=form,
+            form_definition__configuration={
+                "components": [
+                    {
+                        "type": "textfield",
+                        "key": "test",
+                    }
+                ]
+            },
+        )
+        form_variable1 = FormVariableFactory.create(
+            form=form, source=FormVariableSources.user_defined
+        )
+        form_variable2 = FormVariableFactory.create(
+            form=form, source=FormVariableSources.user_defined
+        )
+
+        url = furl(
+            reverse(
+                "api:form-variables",
+                kwargs={"uuid_or_slug": form.uuid},
+            )
+        )
+        url.args["source"] = FormVariableSources.user_defined
+
+        response_user_defined = self.client.get(url.url)
```

Comment:
usually I try to only use `furl` in situations where complicated URL manipulation is needed, but with the django test cases this isn't needed since you can pass a dict for query params:

```suggestion
        response_user_defined = self.client.get(
            reverse("api:form-variables", kwargs={"uuid_or_slug": form.uuid}),
            {"source": FormVariableSources.user_defined},
        )
```